### ISSUE TITLE
Time Range with relative and absolute selection options.

### ISSUE THREAD

**Issue by kalyanixraut** [IBM/carbon-addons-iot-react#664]
### Summary

Hi Team,

I am from TRIRIGA Building Insights development team, we have requirement for a new time range component. More details on the expected functionality and visualization of this component is in the below invision link.

https://ibm.invisionapp.com/share/UBNZD4FX2JE#/screens/319292931

### Specific timeline issues / requests

This new component is required for upcoming early Jan 2020 release.

### You will consider this component complete when...

- [ ] It's available as an export of this library
- [ ] Additional criteria you may have [e.g. Fully responsive, Accessible to WCAG 2.0, etc.]

**Comment by tay1orjones** [comment]
Hi @kalyanixraut 👋thanks for surfacing this request.

This would be a welcome contribution from your team if the design is documented and agreed-upon first by the PAL design team. Please work with your designers to propose this in an issue on the [internal `pal-tracking` repository](https://github.ibm.com/Watson-IoT/pal-tracking).

**Comment by daphoenix** [comment]
The Date and Time range picker for Building Insights for [next release](https://ibm.invisionapp.com/share/KDNZFOF3JBF#/screens/319302295_5-1-UtilRate-Interactive-TimeFilter)

**Additional context**
Add any other context about the component here [e.g. screenshots, code, InVision prototype, Box notes containing details necessary for implementation]

[Invision prototype screen](https://ibm.invisionapp.com/share/KDNZFOF3JBF#/screens/319302295_5-1-UtilRate-Interactive-TimeFilter)
Should be able to :
1. Select a date and time from the calendar widget.
or 
2. Select a start time period from the choice of last 24 hours, 7 days, 30 days, 3 months, 6 months or 12 months.
<img width="613" alt="Screenshot 2019-11-05 at 22 44 33" src="https://user-images.githubusercontent.com/5711817/68248879-f6081d80-001d-11ea-93d3-dd734e1adfa6.png">

**Comment by davidicus** [comment]
closing in favor of issue #854
